### PR TITLE
dynamic sidemenu with submenus

### PR DESCRIPTION
**Pull request by lavahotsnacks**
this is still not done. will continue tomorrow because i need to sleep

this branch needs to be finished first and be merged into develop first, before creating a new branch for other members to work on.

### issues
- parent route is shown in the side-menu but not the children routes
- children routes should be shown below the parent route
- shows the parent route's content on the child route (some logic issues with router-view, need help understanding this)

### CODE REVIEW

**Comment by 29-29** [comment]
> this is still not done. will continue tomorrow because i need to sleep

make it a draft lol if it's still WIP

File: src/components/side-menu/MenuOptions.vue [hunk 1]
```diff
@@ -30,14 +30,30 @@
...
 import { MenuOptionItem } from './types';
 
 const route = useRoute();
+const authStore = useAuthStore();
 const activeOption = ref('');
 
-defineProps<{ menuOptions: MenuOptionItem[] }>();
+const props = defineProps<{ menuOptions: MenuOptionItem[] }>();
+
+const filteredMenuOptions = computed(() => {
```
Comment (29-29) [inline src/components/side-menu/MenuOptions.vue]: I dont think it needs to be computed as the menu options are static values and won't be changed frequently

File: src/pages/admin/ApplicationPage.vue [hunk 2]
```diff
@@ -9,6 +9,7 @@
         <DataTable />
       </q-card-section>
     </q-card>
+    <router-view />
```
Comment (29-29) [inline src/pages/admin/ApplicationPage.vue]: router view is for?

File: src/router/routes.ts
Comment (29-29) [inline src/router/routes.ts:1]: can you add the requiresLogin meta property on the other routes such as the student page?

**Review by 29-29 — COMMENTED** [review]
everything looks good, but apart from that, can we focus on fixing the issue with the sidemenu, such as adding the requiresLogin property on the student page?

File: src/pages/admin/ApplicationPage.vue (hunk 2 above)
Comment (lavahotsnacks) [reply src/pages/admin/ApplicationPage.vue]: that is temporary, to show that the page for the children of the parent route is working.

File: src/router/routes.ts
Comment (lavahotsnacks) [reply src/router/routes.ts:1]: i did add them but forgot to stage those changes, will stage and commit it later.

**Comment by 29-29** [comment]
might wanna rename the pull request lol, as we're only trying to update the sidemenu here?

**Comment by fencim** [comment]
do you have any progress on this task?

**Comment by fencim** [comment]
fixed the conflict

**Comment by lavahotsnacks** [comment]
@29-29 can you please approve this pr, so my devs and i could start working on displaying the application page. also sorry for taking this long, my developing environment broke down 😅

**Comment by 29-29** [comment]
there's an issue with the authentication, i can't check the changes on the site because of it, and I can't approve this PR because I can't see the site

**Review by fencim — APPROVED** [review]

File: src/router/routes.ts [hunk 3]
```diff
@@ -74,14 +74,15 @@ const routes: RouteRecordRaw[] = [
...
           icon: 'person',
         },
       },
       {
-        path: 'booking',
         name: 'booking',
+        path: 'booking',
         component: () => import('pages/students/BookingPage.vue'),
         meta: {
           menu: 'Booking',
```
Comment (29-29) [inline src/router/routes.ts:88]:
```suggestion
          menu: 'Booking',
          requiresLogin: ['student', 'admin']
```

**Pull request merged**